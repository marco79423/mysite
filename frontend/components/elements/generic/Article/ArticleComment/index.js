import ArticleComment from './ArticleComment'

export default ArticleComment
