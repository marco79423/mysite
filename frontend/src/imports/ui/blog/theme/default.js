const colorTable = {
  'cyan-50': '#E0F7FA',
  'cyan-600': '#00ACC1',
  'cyan-700': '#0097A7',
  'cyan-800': '#00838F',
  'cyan-900': '#006064',
  'cyan-A700': '#00B8D4',

  'teal-50':'#E0F2F1',
  'teal-100':'#B2DFDB',
  'teal-200':'#80CBC4',
  'teal-300':'#4DB6AC',
  'teal-400':'#26A69A',
  'teal-500':'#009688',
  'teal-600':'#00897B',
  'teal-700':'#00796B',
  'teal-800':'#00695C',
  'teal-900':'#004D40',
  'teal-A700':'#00BFA5',

  'grey-50': '#FAFAFA',
  'grey-200': '#EEEEEE',
  'grey-300': '#E0E0E0',
  'grey-600': '#757575',
  'grey-700': '#616161',
  'grey-900': '#212121'
}

export default {
  global: {
    fontColor: colorTable['grey-900'],
    link: {
      color: colorTable['cyan-700'],
      hoverColor: colorTable['cyan-900']
    }
  },
  page: {
    background: colorTable['grey-700'],
    header: {
      background: colorTable['teal-400'],
      titleColor: colorTable['grey-50']
    },
    nav: {
      background: colorTable['teal-600'],
      menu: {
        color: '#ECECEC',
        background: colorTable['teal-600'],
        hoverBackground: colorTable['cyan-900'],
        dropdown: {
          background: colorTable['teal-600'],
          borderColor: colorTable['teal-800']
        }
      }
    },
    main: {
      background: colorTable['grey-200'],
      content: {
        article: {
          background: 'white',
          metadata: {
            tip: {
              color: '#FFFFFF',
              background: colorTable['teal-600']
            }
          },
          header: {
            color: colorTable['grey-700']
          },
          block: {
            color: 'white',
            background: colorTable['teal-600'],
            borderColor: '#ccc'
          },
          doctestBlock: {
            color: 'white',
            background: '#6D6D6D',
          },
          note: {
            background: '#ECECEC',
            titleColor: colorTable['teal-800'],
            borderColor: '#ccc'
          },
          table: {
            header: {
              background: colorTable['teal-600'],
            }
          }
        },
        pagination: {
          color: colorTable['grey-200'],
          background: colorTable['teal-700'],
          hoverBackground: colorTable['teal-900'],
          activeBackground: colorTable['teal-900'],
        }
      },
      aside: {
        background: colorTable['grey-200'],
        darkerBackground: colorTable['grey-600'],

        section: {
          titleColor: colorTable['grey-700'],
          borderColor: colorTable['grey-200']
        }
      }
    },
    footer: {
      background: colorTable['teal-600'],
      copyrightColor: '#ECECEC',
    }
  }
}