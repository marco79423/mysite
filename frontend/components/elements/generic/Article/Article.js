import React from 'react'
import styled from 'styled-components'
import {LikeCoinButton} from '@paji-sdk/browser'

import TitleLink from '../TitleLink'
import RstContent from '../RstContent'
import SocialShare from './SocialShare'
import Metadata from './Metadata'
import ArticleComment from './ArticleComment'


const Base = styled.article`
  background: ${props => props.theme.page.main.content.article.background};
  padding: 32px;
`

const Header = styled.header`
  h1 {
    margin: 3px 0;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
  }
`

class Article extends React.PureComponent {

  renderSummary = () => {
    return <RstContent content={this.props.article.summary}/>
  }

  renderContent = () => {
    const {article, commentConfig} = this.props
    return [
      <RstContent key="article-content" content={article.content}/>,
      <LikeCoinButton
        key="likeCoin"
        creatorLikeID={'marco79423'}
        url={article.url}
        style={{marginTop: 48, height: 212, width: '100%'}}/>,
      <SocialShare key="social-share" config={{
        title: article.title,
        shareUrl: article.url,
      }}/>,
      <ArticleComment key="comment" config={commentConfig}/>
    ]
  }

  render() {
    const {article, summaryMode} = this.props
    return (
      <Base>
        <Header>
          <h1><TitleLink href={article.path}>{article.title}</TitleLink></h1>
          <Metadata
            categories={article.categories}
            chickenCount={article.chickenCount}
            date={article.date}
            modifiedDate={article.modifiedDate}/>
        </Header>

        {summaryMode ? this.renderSummary() : this.renderContent()}
      </Base>
    )
  }
}


export default React.memo(Article)
