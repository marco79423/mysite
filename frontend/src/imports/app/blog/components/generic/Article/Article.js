import React from 'react'
import PropTypes from 'prop-types'
import ImmutablePropTypes from 'react-immutable-proptypes'
import styled from 'styled-components'

import TitleLink from '../TitleLink'
import Loading from '../Loading'
import RstContent from '../RstContent'
import SocialShare from './SocialShare'
import Metadata from './Metadata'
import ArticleComment from './ArticleComment'


const Base = styled.article`
  background: ${props => props.theme.page.main.content.article.background};
  padding: 32px;
`

const Header = styled.header`
  h1 {
    margin: 3px 0;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
  }
`

export default class Article extends React.PureComponent {
  static PropTypes = {
    summaryMode: PropTypes.bool,
    article: ImmutablePropTypes.contains({
      slug: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
      categories: ImmutablePropTypes.listOf(ImmutablePropTypes.contains({
        slug: PropTypes.string.isRequired,
        name: PropTypes.string.isRequired
      })).isRequired,
      chickenCount: PropTypes.number.isRequired,
      date: PropTypes.instanceOf(Date).isRequired,
      modifiedDate: PropTypes.instanceOf(Date),
      content: PropTypes.any.isRequired,
      summary: PropTypes.string.isRequired
    }),
    socialConfig: ImmutablePropTypes.map,
    commentConfig: ImmutablePropTypes.map
  }

  renderHeader = () => {
    const {article} = this.props
    return (
      <Header>
        <h1><TitleLink to={`/articles/${article.get('slug')}/`}>{article.get('title')}</TitleLink></h1>
        {this.renderMetadata()}
      </Header>
    )
  }

  renderMetadata = () => {
    const {article} = this.props
    return (
      <Metadata
        categories={article.get('categories')}
        chickenCount={article.get('chickenCount')}
        date={article.get('date')}
        modifiedDate={article.get('modifiedDate')} />
    )
  }

  renderSummary = () => {
    return <RstContent content={this.props.article.get('summary')}/>
  }

  renderContent = () => {
    const {article, socialConfig, commentConfig} = this.props
    return [
      <RstContent key="article-content" content={article.get('content')}/>,
      <SocialShare  key="social-share" config={socialConfig}/>,
      <ArticleComment key="comment" config={commentConfig}/>
    ]
  }

  render () {
    const {article, summaryMode} = this.props
    if (!article) {
      return (
        <Base>
          <Loading/>
        </Base>
      )
    }

    return (
      <Base>
        {this.renderHeader()}
        {summaryMode ? this.renderSummary() : this.renderContent()}
      </Base>
    )
  }
}

