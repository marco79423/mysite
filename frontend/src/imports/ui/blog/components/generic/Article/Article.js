import React from 'react'
import styled from 'styled-components'
import {LikeCoinButton} from '@paji-sdk/browser'

import TitleLink from '../TitleLink'
import Loading from '../Loading'
import RstContent from '../RstContent'
import SocialShare from './SocialShare'
import Metadata from './Metadata'
import ArticleComment from './ArticleComment'


const Base = styled.article`
  background: ${props => props.theme.page.main.content.article.background};
  padding: 32px;
`

const Header = styled.header`
  h1 {
    margin: 3px 0;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
  }
`

export default class Article extends React.PureComponent {

  renderHeader = () => {
    const {article} = this.props
    return (
      <Header>
        <h1><TitleLink to={`/articles/${article.slug}/`}>{article.title}</TitleLink></h1>
        {this.renderMetadata()}
      </Header>
    )
  }

  renderMetadata = () => {
    const {article} = this.props
    return (
      <Metadata
        categories={article.categories}
        chickenCount={article.chickenCount}
        date={article.date}
        modifiedDate={article.modifiedDate}/>
    )
  }

  renderSummary = () => {
    return <RstContent content={this.props.article.summary}/>
  }

  renderContent = () => {
    const {article, socialConfig, commentConfig} = this.props
    return [
      <RstContent key="article-content" content={article.content}/>,
      <LikeCoinButton
        key="likeCoin"
        creatorLikeID={'marco79423'}
        url={socialConfig?.shareUrl}
        style={{marginTop: 48, height: 212, width: '100%'}}/>,
      <SocialShare key="social-share" config={socialConfig}/>,
      <ArticleComment key="comment" config={commentConfig}/>
    ]
  }

  render() {
    const {article, summaryMode} = this.props
    if (!article) {
      return (
        <Base>
          <Loading/>
        </Base>
      )
    }

    return (
      <Base>
        {this.renderHeader()}
        {summaryMode ? this.renderSummary() : this.renderContent()}
      </Base>
    )
  }
}

