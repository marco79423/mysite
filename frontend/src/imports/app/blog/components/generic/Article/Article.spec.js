import Immutable from 'immutable'
import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import {ThemeProvider} from 'styled-components'

import Article from './Article'
import theme from '../../../theme/default'

test('It should render Article component correctly', () => {
  const article = Immutable.fromJS({
    slug: 'slug',
    title: 'title',
    categories: [{
      slug: 'slug',
      name: 'name'
    }],
    date: new Date(2017, 6, 9),
    chickenCount: 1,
    content: 'content',
    rawSummary: 'raw_summary'
  })
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <Article article={article}/>
    </ThemeProvider>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('It should render Article component correctly when the article is null', () => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <Article article={null}/>
    </ThemeProvider>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
