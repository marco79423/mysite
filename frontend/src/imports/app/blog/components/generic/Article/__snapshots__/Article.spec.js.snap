// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render Article component correctly 1`] = `
.c3 {
  color: #0097A7;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c3:hover {
  color: #006064;
}

.c2 {
  font-size: 2rem;
}

.c10 {
  margin-top: 16px;
  font-size: 1.2rem;
  line-height: 1.6rem;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 1rem;
}

.c11 > div:not(:first-child) {
  margin-left: .5rem;
}

.c4 {
  font-size: 0.9rem;
  line-height: 1.6rem;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c5 {
  display: inline;
}

.c5:not(:first-child) {
  margin-left: 8px;
}

.c6 {
  display: inline;
}

.c7 {
  margin: 0;
  display: inline;
}

.c7:not(:first-child):before {
  content: "/";
  margin: 0 3px;
}

.c8 {
  margin-bottom: -5px;
}

.c8:hover {
  -webkit-transform: rotate(-10deg);
  -ms-transform: rotate(-10deg);
  transform: rotate(-10deg);
}

.c9 {
  display: none;
  position: absolute;
  line-height: 1.2em;
  padding: 4px;
  border-radius: 3px;
  background: #00897B;
  color: #FFFFFF;
  -webkit-transition: opacity .3s ease-out;
  transition: opacity .3s ease-out;
}

.c9:before {
  content: "";
  position: absolute;
  width: 0;
  height: 0;
  left: -15px;
  top: 4px;
  border: 8px solid transparent;
  border-right-color: #00897B;
}

.c0 {
  background: white;
  padding: 32px;
}

.c1 h1 {
  margin: 3px 0;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

@media (max-width:800px) {
  .c9 {
    display: none;
  }
}

<article
  className="c0"
>
  <header
    className="c1"
  >
    <h1>
      <a
        className="c2 c3"
        onClick={[Function]}
        style={Object {}}
      >
        title
      </a>
    </h1>
    <ul
      className="c4"
    >
      <li
        className="c5"
      >
        分類：
        <ul
          className="c6"
        >
          <li
            className="c7"
          >
            <a
              className="c3"
              onClick={[Function]}
              style={Object {}}
            >
              name
            </a>
          </li>
        </ul>
      </li>
      <li
        className="c5"
      >
        字數：
        <img
          alt="雞"
          className="c8"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          src={
            Object {
              "default": Object {},
            }
          }
        />
         x 
        1
        <span
          className="c9"
        >
          雞數：計算文長的常見計量單位，一般而言數字大小與文章長度呈正相關
        </span>
      </li>
      <li
        className="c5"
      >
        發表時間：
        2017/07/09
      </li>
    </ul>
  </header>
  <div
    className="rst-content c10"
    dangerouslySetInnerHTML={
      Object {
        "__html": "content",
      }
    }
  />
  <div
    className="c11"
  />
  <div />
</article>
`;

exports[`It should render Article component correctly when the article is null 1`] = `
.c1 {
  margin-top: 10%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  display: inline-block;
  width: 90px;
  height: 90px;
  background: url() no-repeat center;
  background-size: 90px 90px;
  border: 0.1em solid #FFFFFF;
  box-shadow: 0 1px 4px rgba(0,0,0,0.15);
  -webkit-animation-name: bQMHby;
  animation-name: bQMHby;
  -webkit-animation-duration: 5s;
  animation-duration: 5s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  z-index: 1000;
}

.c3 {
  margin-top: 2rem;
  font-size: 2rem;
  text-align: center;
}

.c0 {
  background: white;
  padding: 32px;
}

<article
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    />
    <div
      className="c3"
    >
      正在努力讀取中…
    </div>
  </div>
</article>
`;
